lmsweb/views.py

```diff
@@ -62,11 +63,22 @@ def load_user(user_id):
     return User.get_or_none(id=user_id)
 
 
+def admins_only(func):
+    """Decorator enforrcing access for admins only"""
+
+    # Must have @wraps to work with endpoints.
+    @wraps(func)
+    def wrapper(*args, **kwargs):
+        if not current_user.role.is_admin:
+            return fail(403, "This user has no permissions to view this page.")
+        else:
+            return func(*args, **kwargs)
+
+    return wrapper
+
+
 def fail(status_code: int, error_msg: str):
-    return abort(
-        status_code,
-        jsonify({'status': 'failed', 'msg': error_msg}),
-    )
+    return abort(status_code, {'status': 'failed', 'msg': error_msg})
```

Comment:
We should use `jsonify` for the frontend

Reply:
Correct, this returned an invalid response - this is now fixed